### ISSUE TITLE
Creating surface mesh from 3d volume.

### ISSUE THREAD

**Issue by JBum21** [pyvista/pyvista-support#402]
Hi,
I have 3d volumetric datasets represented in Numpy arrays. These datasets are all binarized.

I am hoping to create a smooth surface that wraps around all of the individual components of my dataset. 

<img width="1011" alt="Screen Shot 2021-04-12 at 3 36 59 PM" src="https://user-images.githubusercontent.com/70919881/114451469-471e2880-9ba5-11eb-98a0-ae552a04b3e9.png">

One example can be seen below, but I can't find the associated code with this example from [#231](https://github.com/pyvista/pyvista/pull/231)
<img width="809" alt="Screen Shot 2021-04-12 at 3 38 34 PM" src="https://user-images.githubusercontent.com/70919881/114451460-44233800-9ba5-11eb-80a9-5aa2f36718a7.png">

Any help or suggestions would be much appreciated. 

I've tried rendering these volumes by computing the radius/centerlines of the individual processes and then creating splines/tube wraps, but the data end up looking quite odd.

Thanks

**Comment by JBum21** [comment]
Hi,
I have 3d volumetric datasets represented in Numpy arrays. These datasets are all binarized.

I am hoping to create a smooth surface that wraps around all of the individual components of my dataset. 

<img width="1011" alt="Screen Shot 2021-04-12 at 3 36 59 PM" src="https://user-images.githubusercontent.com/70919881/114451469-471e2880-9ba5-11eb-98a0-ae552a04b3e9.png">


One example can be seen below, but I can't find the associated code with this example from [#231](https://github.com/pyvista/pyvista/pull/231)
<img width="809" alt="Screen Shot 2021-04-12 at 3 38 34 PM" src="https://user-images.githubusercontent.com/70919881/114451460-44233800-9ba5-11eb-80a9-5aa2f36718a7.png">

Any help or suggestions would be much appreciated. 

I've tried rendering these volumes by computing the radius/centerlines of the individual processes and then creating splines/tube wraps, but the data end up looking quite odd.

Thanks

**Comment by JBum21** [comment]
Ok - so I managed to figure out the beginnings of a solution to this specific problem. I'm posting the beginnings of the results here. When I optimize them, I will post more links in the future.

What I learned is that I can create an isosurface (surface mesh) from my scalar volume (3D Numpy array, binarized in my case) using a marching cubes algorithm. I tried using Scipy.measure.marching_cubes, but Pyvista kept telling me that it was producing non-manifold surfaces. 

As such, I used the [PyMCubes.marching_cubes](https://github.com/pmneila/PyMCubes) package to create my isosurface. 

<img width="1570" alt="Screen Shot 2021-04-13 at 1 37 01 PM" src="https://user-images.githubusercontent.com/70919881/114596233-5fea1500-9c5d-11eb-8a3e-792b67169993.png">

I'm going to continue experimenting with different subdivisions/smoothing operations to find some ideal settings.
